app/labor/rate_limit_checker.rb

```diff
@@ -9,22 +9,13 @@ def initialize(user = nil)
     @user = user
   end
 
-  class UploadRateLimitReached < StandardError; end
-  class DailyFollowAccountLimitReached < StandardError; end
+  UploadRateLimitReached = Class.new(StandardError)
+  DailyFollowAccountLimitReached = Class.new(StandardError)
 
   def limit_by_action(action)
-    result = case action
-             when "comment_creation"
-               user.comments.where("created_at > ?", 30.seconds.ago).size > 9
-             when "published_article_creation"
-               user.articles.published.where("created_at > ?", 30.seconds.ago).size > 9
-             when "image_upload"
-               Rails.cache.read("#{user.id}_image_upload").to_i > 9
-             when "follow_account"
-               user_today_follow_count > self.class.daily_account_follow_limit
-             else
-               false
-             end
+    check_method = "check_#{action}_limit"
+    result = respond_to?(check_method) ? public_send(check_method) : false
```

Comment:
Not a big fan of this ternary. Originally I had 

```ruby
result = public_send(check_method) if respond_to?(check_method)
```

but the specs expect `false` explicitly, not a falsey value.  I could have changed them to use the `be_falsey` matcher, but since this is still a WIP PR I decided to wait for feedback on this.

Reply:
Looks fine